Factory.Cargo/CargoFactoryApi.cs

```diff
@@ -0,0 +1,36 @@
+using Factory.Cargo.Factory;
+using Factory.Cargo.TransportationFactory;
+
+namespace Factory.Cargo;
+
+public class CargoFactoryApi
+{
+    private Dictionary<string, ITransportationFactory> _transportationFactories = new();
+
+    public void AddBaseFactories()
```

Comment:
why you did not move it into constructor? 
you user must know to call this method before every usage.

Reply:
I have enabled user to figure out this feature by optional flag on Ctor.
Please checkout [fb9738f].